api/app/signals/apps/signals/v1/views.py

```diff
@@ -53,11 +57,112 @@ def get_object(self):
         return obj
 
 
-class PrivateSignalViewSet(DatapuntViewSet,
+class PublicSignalViewSet(mixins.CreateModelMixin,
+                          DetailSerializerMixin,
+                          mixins.RetrieveModelMixin,
+                          viewsets.GenericViewSet):
+    queryset = Signal.objects.all()
+    serializer_class = PublicSignalCreateSerializer
+    serializer_detail_class = PublicSignalSerializerDetail
+    lookup_field = 'signal_id'
+
+
+class GeneratePdfView(LoginRequiredMixin, SingleObjectMixin, PDFTemplateView):
+    object = None
+    pk_url_kwarg = 'signal_id'
+    queryset = Signal.objects.all()
+
+    template_name = 'signals/pdf/print_signal.html'
+    extra_context = {'now': timezone.datetime.now(), }
+
+    def get_context_data(self, **kwargs):
+        self.object = self.get_object()
+        self.pdf_filename = 'SIA-{}.pdf'.format(self.object.pk)
+        rd_coordinates = self.object.location.get_rd_coordinates()
+        bbox = '{},{},{},{}'.format(
+            rd_coordinates.x - 340.00,
+            rd_coordinates.y - 125.00,
+            rd_coordinates.x + 340.00,
+            rd_coordinates.y + 125.00,
+        )
+        return super(GeneratePdfView, self).get_context_data(bbox=bbox)
+
+
+class PublicSignalAttachmentsViewSet(mixins.CreateModelMixin, viewsets.GenericViewSet):
+    serializer_class = SignalAttachmentSerializer
+    is_public = True
+    lookup_field = 'signal_id'
+    lookup_url_kwarg = 'signal_id'
+    queryset = Signal.objects.all()
+
+
+""" PRIVATE endpoints """
+
+
+class PrivateSignalViewSetBase:
```

Comment:
I think this is not the way to solve the problem we have with the permissions. It feels unfinished and buggy. Also queries needs to be performed twice to avoid getting 404, why??? This needs to be refactored in a better and nicer solution.

Reply:
Like discussed before